app/src/main/res/layout/activity_main.xml

```diff
@@ -4,15 +4,240 @@
+        app:layout_constraintBottom_toTopOf="@id/button_subtraction"
+        app:layout_constraintEnd_toEndOf="@id/button_subtraction"
+        app:layout_constraintStart_toStartOf="@id/button_subtraction" />
+
+    <Button
+        android:id="@+id/button_one"
+        style="@style/button_style"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintHeight_default="wrap"
+        android:text="@string/activity_main_calculator_one_text"
+        app:layout_constraintBottom_toTopOf="@id/button_zero"
+        app:layout_constraintEnd_toEndOf="@id/button_zero"
+        app:layout_constraintStart_toStartOf="@id/button_zero" />
+
+    <Button
+        android:id="@+id/button_two"
+        style="@style/button_style"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintHeight_default="wrap"
+        android:text="@string/activity_main_calculator_two_text"
+        app:layout_constraintBottom_toTopOf="@id/button_clean"
+        app:layout_constraintEnd_toEndOf="@id/button_clean"
+        app:layout_constraintStart_toStartOf="@id/button_clean" />
+
+    <Button
+        android:id="@+id/button_three"
+        style="@style/button_style"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintHeight_default="wrap"
+        android:text="@string/activity_main_calculator_three_text"
+        app:layout_constraintBottom_toTopOf="@id/button_equals"
+        app:layout_constraintEnd_toEndOf="@id/button_equals"
+        app:layout_constraintStart_toStartOf="@id/button_equals" />
+
+    <Button
+        android:id="@+id/button_subtraction"
+        style="@style/button_style"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintHeight_default="wrap"
+        android:text="@string/activity_main_calculator_subtraction_text"
+        app:layout_constraintBottom_toTopOf="@id/button_add"
+        app:layout_constraintEnd_toEndOf="@id/button_add"
+        app:layout_constraintStart_toStartOf="@id/button_add" />
+
+    <Button
+        android:id="@+id/button_zero"
+        style="@style/button_style"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintHeight_default="wrap"
+        android:layout_marginHorizontal="@dimen/margin_between_button_xsmall"
+        android:text="@string/activity_main_calculator_zero_text"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toStartOf="@id/button_clean"
+        app:layout_constraintStart_toStartOf="parent" />
+
+    <Button
+        android:id="@+id/button_clean"
+        style="@style/button_style"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintHeight_default="wrap"
+        android:layout_marginHorizontal="@dimen/margin_between_button_xsmall"
```

Comment:
you use this margin in several buttons, please create a new style for this if it is possible

Reply:
Done!